crates/test-macros/src/lib.rs

```diff
@@ -0,0 +1,93 @@
+//! Wasmtime test macro.
+//!
+//! This macro is a helper to define tests that exercise multiple configuration
+//! combinations for Wasmtime. Currently, only compiler strategies are
+//! supported.
+//!
+//! Usage
+//!
+//! #[wasmtime_test(strategies(Cranelift, Winch))]
+//! fn my_test(config: &mut Config) -> Result<()> {
+//!    Ok(())
+//! }
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{
+    parenthesized, parse::Parse, parse_macro_input, Ident, ItemFn, Result, ReturnType, Token,
+};
+
+#[proc_macro_attribute]
+pub fn wasmtime_test(attrs: TokenStream, item: TokenStream) -> TokenStream {
+    let mut strategies: Vec<(String, Ident)> = vec![];
+
+    let parser = syn::meta::parser(|meta| {
+        if meta.path.is_ident("strategies") {
+            let v = meta.input;
+            let content;
+            parenthesized!(content in v);
+            strategies = content
+                .parse_terminated(Ident::parse, Token![,])?
+                .into_iter()
+                .map(|v| (v.to_string(), v))
+                .collect();
+
+            if strategies.len() > 2 {
+                return Err(meta.error("Expected at most 2 strategies"));
+            }
+
+            if strategies.is_empty() {
+                return Err(meta.error("Expected at least 1 strategy"));
+            }
+
+            Ok(())
+        } else {
+            Err(meta.error("Unsupported attributes"))
+        }
+    });
+
+    parse_macro_input!(attrs with parser);
+
+    match expand(&strategies, parse_macro_input!(item as ItemFn)) {
```

Comment:
One possible concern about `ItemFn` here is that in the limit the entire test crate will be tagged with `#[wasmtime_test]` and that means the entire crate is parsed through `syn` and emitted back out. Technically the only needed part here is the function signature. 

I'm not sure if it's worth it, but this could try to be a bit clever where it parses `syn::ItemFn` manually up to the `block` and then keeps the `block` as an opaque `TokenStream` (since it's known that it's brace-delimited anyway).